service/svcutil/svcutil_windows.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 PaperCut Software http://www.papercut.com/
+// Use of this source code is governed by an MIT or GPL Version 2 license.
+// See the project's LICENSE file for more information.
+package svcutil
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+// SetServiceToRestart sets the service named name to automatically restart after waiting for a duration specified by
+// delay when the service fails, and the time after which to reset the service failure count to zero if there are no
+// failures, in seconds. Specify systemcall.INFINITE to indicate that service failure count should never be reset. The
+// service has to already be registered in the SCM.
+func SetServiceToRestart(name string, delay time.Duration, resetPeriod uint32) error {
```

Comment:
Probably `resetPeriod` should be `time.Duration`, and set INFINITY internally if the value is negative.